charts/tekton-pipelines/templates/_snippets.tpl

```diff
@@ -283,6 +283,36 @@ finally:
     - deploy
 {{- end }}
 
+# ┌──────────────────────────────────────────────────────────────────────────────┐
+# │ pre deploy reusable snippet in various pipelines                             │
+# │                                                                              │
+# └──────────────────────────────────────────────────────────────────────────────┘
+{{- define "pipeline.preDeploy" -}}
+- name: pre-deploy
+  taskRef:
+    name:  {{ .name }}
+  resources:
+    inputs:
+      - name: app
+        resource: app
+  params:
+    - name: application
+      value: "$(params.application)"
+    - name: project
+      value: "$(params.project)"
+    - name: namespace
+      value: "$(params.namespace)"
+    - name: sha
+      value: "$(params.sha)"
+    - name: environment
+      value: "$(params.environment)"
+  workspaces:
+    - name: source
+      workspace: source
+  runAfter:
+    - kustomize
+{{- end }}
+
```

Comment:
expose all params as envs to the task steps
https://github.com/saritasa-nest/usummit-kubernetes-aws/blob/5f12a2a874bbcbf8e3b098bc687965770dd78e2e/apps/addons/tekton-pipelines/build-pipeline/values.yaml#L429-L443

that helps use ${envName}, instead of tekton-based replacement via $(params.name) in the code of the step

Reply:
![image](https://github.com/saritasa-nest/saritasa-devops-helm-charts/assets/26448672/31dc8bd5-b679-4877-baf3-2b92d38b6155)